osu.Game/Skinning/SkinReloadableDrawable.cs

```diff
@@ -56,6 +68,22 @@ protected virtual void SkinChanged(ISkinSource skin)
         {
         }
 
+        private void onChange()
+        {
+            // schedule required to avoid calls after disposed.
+            // note that this has the side-effect of components only performing a skin change when they are alive.
+            pendingSkinChange?.Cancel();
+            pendingSkinChange = Scheduler.Add(skinChanged);
+        }
+
+        private void skinChanged()
+        {
+            SkinChanged(CurrentSkin);
+            OnSkinChanged?.Invoke();
+
+            pendingSkinChange = null;
```

Comment:
Nulling a field in a callback gives me the heebie-jeebies. I don't really know what the threading on this is but I'm imagining a scenario involving threading wherein this nulls the wrong `pendingSkinChange` because it's not checking that `pendingSkinChange` hasn't changed since when the callback was scheduled.

It would *probably* matter less if `skinChanged` was `AddOnce()`'d rather than `Add()`ed, since the former is more of a synchronisation point than a direct command (as in, with `AddOnce()` you shouldn't really need to do much of this callback tracking, the only thing that matters is that the thing fires next frame).

Reply:
Since `skinChanged()` takes no parameters and only uses fields instead this is likely fine. Was mostly just talking out loud here.